<-- http://alan.cs.gsu.edu/NGS/?q=software-->

* Funded Research Projects
* Request new password
#  **These software tools may be freely used for all non-commercial purposes. **
(c) Copyright, 2019 by Professor Alexander Zelikovsky 
Department of Computer Science, Georgia State University 
Atlanta, GA 30303 
CliqueSNV Quasispecies reconstruction from NGS reads
QUASIM QUAsispecies SIMulator software for viral population evolution with immune response
IsoEM 1.1.4 The IsoEM package can be used to infer isoform and gene expression levels from high-throughput transcriptome sequencing (RNA-Seq) data
xpathway : A set of tools that compares metabolic pathway activity analyzing mapping of contigs assembled from RNA-Seq reads to KEGG pathways. The XPathway analysis of pathway activity is based on expectation maximization and topological properties of pathway graphs.
2SNV Quasispecies reconstruction from long reads
ScaffMatch: Scaffolding Algorirthm Based on Maximum Weight Matching
Pooling: Computational framework for next-generation sequencing of heterogeneous viral populations using combinatorial pooling
VGA: Viral Genome Assembler is a method for accurate assembly of a heterogeneous viral population coupled with a high-fidelity sequencing protocol able to eliminate errors from sequencing data.
kGEM k-Genotype Expectation Maximization algorithm for Reconstructing a Viral population from Single-Amplicon reads
VirA/VirA-MCF reconstruct viral quasipecies and estimate their frequencies from amplicon reads.
IsoEM: Infers isoform and gene expression levels from high-throughput transcriptome sequencing (RNA-Seq) data.
KEC Error correction for pyrosequencing reads (454 Life Sciences and Ion Torrent)
_MaLTA_ : Transcriptome assembly and quantification from RNA-Seq reads
ViSpA: Viral Spectrum Assembler implements a novel viral assembling and frequency estimation methods. This software uses a simple error correction, viral variants assembling based on maximum-bandwidth paths in weighted read graphs and frequency estimation via Expectation Maximization on all reads.
2SNP: SCALABLE PHASING METHOD BASED ON 2-SNP HAPLOTYPES. New phasing is based on statistically significant LD and deviation from Hardy-Weinberg equilibrium. On datasets across 69 regions from HapMap 2SNP is 3-4 orders of magnitude faster and usually outperforms HAPLOTYPER, GERBIL and matches PHASE.
**Please, contact alexz@cs,gsu,edu regarding older software tools below: **
MetNetAligner: Web service tool for
* matching metabolic pathways/networks
* identifying pathway holes (missing enzymes) and suggesting plausible candidates
* and finding network motifs
DACS: Disease Association Combinatorial Search software searches for statistically significant multi-SNP combinations associated with a disease.
Tagging: Tag Selection based on SNP Multivariate Linear Regression.
Disease Association: This project explores disease susceptibility prediction on genotype/haplotype data.
Trio Phasing: Frequently case/control genotype data represents family trios consisting of two parents and one offspring. The trio phasing project develops a satisfactory method for phasing family trio data.
Georgia State University NGS Research Group