<-- https://www.ebi.ac.uk/seqdb/confluence/display/JDSAT/References-->

* Skip to content
* Skip to breadcrumbs
* Skip to header menu
* Skip to action menu
* Skip to quick search
*   * Quick Search
* Online Help 
* Keyboard Shortcuts 
* Feed Builder 
* What’s new 
* Available Gadgets 
* About Confluence 
*   *   *   * Log in 
Job Dispatcher Sequence Analysis Tools
* Your Privacy, Help & Support
*     * A_t_tachments (0) 
* Page History 
* Page Information 
* Resolved comments 
* View in Hierarchy 
* View Source 
* Export to PDF 
* Export to Word 
2. Job Dispatcher Sequence Analysis Tools Home
Skip to end of banner
*   * Jira links
Go to start of banner
Skip to end of metadata
* Created by  Karyn Duggan, last modified by  Fabio Madeira on May 02, 2019
Go to start of metadata
Madeira, F., Park, YM., Lee, J., Buso, N., Gur, T., Madhusoodanan, N., Basutkar, P., Tivey, ARN., Potter, SC., Finn, RD. and Lopez, R. (2019). 
**The EMBL-EBI search and sequence analysis tools APIs in 2019**. _Nucleic Acids Research_, gkz268.  
EuropePMC: _ 30976793 _
Madeira, F., Madhusoodanan, N., Lee, J., Tivey, A. R. N., and Lopez, R. (2019). 
**Using EMBL‐EBI services via web interface and programmatically via web services.**  
_Current Protocols in Bioinformatics_, e74. 
Chojnacki S, Cowley A, Lee J, Foix A, Lopez R. (2017) 
**Programmatic access to bioinformatics tools from EMBL-EBI update: 2017.**  
Nucleic Acids Res Volume (2017) p. 
Li W., Cowley A., Uludag M., Gur T., McWilliam T., Squizzato S., Park YM., Buso N., Lopez R. (2015) 
**The EMBL-EBI bioinformatics web and programmatic tools framework. 
**Nucleic acids research 43 (W1) : W580-4 
McWilliam H., Li W., Uludag M., Squizzato S., Park YM., Buso N., Cowley A., Lopez R. (2013) 
**Analysis Tool Web Services from the EMBL-EBI.**  
Nucleic acids research 41 (Web Server issue) : W597-600 
Goujon, M., McWilliam, H., Li, W., Valentin, F., Squizzato, S., Paern, J. and Lopez, R. (2010) 
**A new bioinformatics analysis tools framework at EMBL-EBI. **  
_Nucleic Acids Res_ 38 (Web Server issue) : W695-9 
McWilliam H., Valentin F., Goujon M., Li W., Narayanasamy M., Martin J., Miyar T. and Lopez R. (2009) 
**Web Services at the European Bioinformatics Institute**  
Nucleic Acids Research 37: W6-W10. 
_ PMID:  19435877 
DOI:  10.1093/nar/gkp302 _
Labarga A., Valentin F., Andersson M. and Lopez R. (2007) 
**Web Services at the European Bioinformatics Institute**  
Nucleic Acids Research 35: W6-W11. 
PMID:  17576686 Abstract 
Labarga A., Pilai S., Valentin F., Anderson M. and Lopez R. (2005) 
**Web services at the European Bioinformatics Institute**  
Pillai S., Silventoinen V., Kallio K., Senger M., Sobhany S., Tate J., Velankar S., Golovin A., Henrick K., Rice P., Stoehr P. and Lopez R. (2005) 
**SOAP-based services provided by the European Bioinformatics Institute**  
Nucleic Acids Research 33: W25-W28. 
PMID:  15980463 
Harte N., Silventoinen V., Quevillon E., Robinson S., Kallio K., Fustero X., Patel P., Jokinen P. and Lopez R. (2004) 
**Public web-based services from the European Bioinformatics Institute**  
Nucleic Acids Research 32: W3-W9. 
PMID:  15215339 
# Tool specific references
Sievers F., Wilm A., Dineen D., Gibson T.J., Karplus K., Li W., Lopez R., McWilliam H., Remmert M., Söding J., Thompson J.D. and Higgins D.G. (2011) 
**Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega.**  
Mol. Syst. Biol. 7:539 
PMID:  _21988835_ 
DOI: _ 10.1038/msb.2011.75 _
Lassmann T., Frings O. and Sonnhammer E.L. (2009) 
**Kalign2: high-performance multiple alignment of protein and nucleotide sequences allowing external features.**  
Nucleic Acids Research 37(3):858-65 
_ _DOI 10.1093/nar/gkn1006 _
Lassmann T., Sonnhammer E.L. (2005) 
**Kalign--an accurate and fast multiple sequence alignment algorithm.**  
BMC Bioinformatics 6:298 
_ PubMed:  16343337 
_ _ DOI  10.1186/1471-2105-6-29 _
Lassmann T., Sonnhammer E.L. (2006) 
**Kalign, Kalignvu and Mumsa: Web servers for multiple sequence alignment**  
Nucleic Acids Research 34 (Web Server issue)::596-9 
_ PubMed:  16845078 
_ _ DOI  10.1093/nar/gkl191 _
Katoh K. , Standley D.M. (2013) 
**MAFFT multiple sequence alignment software version 7: improvements in performance and usability.**  
Molecular biology and evolution 30 (4) :772-780 
PubMed:  23329690 
Edgar, R.C. (2004) 
**MUSCLE: multiple sequence alignment with high accuracy and high throughput**  
Nucleic Acids Research 32(5):1792-1797 
Edgar, R.C. (2004) 
**MUSCLE: a multiple sequence alignment method with reduced time and space complexity**  
_BMC Bioinformatics_ 19;5:113 
Brown N.P., Leroy C. and Sander C. (1998) 
**MView: a web-compatible database search or multiple alignment viewer**  
Löytynoja A. and Goldman N. (2005) 
**An algorithm for progressive multiple alignment of sequences with insertions**  
Proc. Natl. Acad. Sci. USA 102:10557-10562 
PubMed:  16000407 
Notredame C., Higgins D.G. and Heringa J. (2000)** 
**T-Coffee: A novel method for fast and accurate multiple sequence alignment.**  
**Journal of Molecular Biology 302(1):205-17 
Rice P., Longden I. and Bleasby A. (2000) 
**EMBOSS: The European Molecular Biology Open Software Suite**  
Trends Genet. 16(6)276-277 
Waterman M.S. and Eggert M. (1987) 
**A new algorithm for best subsequence alignments with application to tRNA-rRNA comparisons**  
J. Mol. Biol. 197(4):723-8 
Needleman S.B. and Wunsch C.D. (1970) 
**A general method applicable to the search for similarities in the amino acid sequence of two proteins**  
J. Mol. Biol. 48(3):443-53 
Myers E.W. and Miller W. (1988) 
**Optimal alignments in linear space**  
Comput. Appl. Biosci. 4(1):11-7 
Smith T.F. and Waterman M.S. (1981) 
**Identification of common molecular subsequences**  
J. Mol. Biol. 147(1):195-7 
Clamp M., Durbin R. and Birney E. (2004) 
**GeneWise and GenomeWise**  
Genome Research 14(5):988-95 
Birney E., Durbin R. 
**Using GeneWise in the Drosophila annotation experiment. **  
Genome research 10 (4) :547-8 
Ettwiller L., Paten B., Souren M., Loosli F., Wittbrodt J., and Birney E. 
** The discovery, positioning and verification of a set of transcription-associated motifs in vertebrates **  
Genome biology 6 (12) :R104 
Jareborg N., Birney E. and Durbin R. (1999) 
**Comparative analysis of noncoding regions of 77 orthologous mouse and human gene pairs**  
Genome Research 14(5):988-95 
Brendel V., Bucher P., Nourbakhsh I.R., Blaisdell B.E. and Karlin S. (1992) 
**Methods and algorithms for statistical analysis of protein sequences.**  
Proc. Natl. Acad. Sci. USA 89(6):2002-6 
Larkin M.A., Blackshields G., Brown N.P., Chenna R., McGettigan P.A., McWilliam H., Valentin F., Wallace I.M., Wilm A., Lopez R., Thompson J.D., Gibson T.J. and Higgins D.G. (2007) 
**Clustal W and Clustal X version 2.0.**  
Sokal, R. and Michener, C. (1958) 
**A statistical method for evaluating systematic relationships**  
University of Kansas Science Bulletin 38:1409–1438.
Sneath P.H.A. and Sokal R.R. (1973) 
**Numerical taxonomy — the principles and practice of numerical classification.**  
(W. H. Freeman: San Francisco.)
Kimura, M. (1980) 
**A simple method for estimating evolutionary rates of base substitutions through comparative studies of nucleotide sequences.**  
Journal of Molecular Evolution 16:111–120. 
Saitou N. and Nei. M. (1987) 
**The neighbor-joining method: a new method for reconstructing phylogenetic trees.**  
Mol. Evol. Biol 4(4):406-425 
R.D. Finn, J. Clements, W. Arndt, B.L. Miller, T.J. Wheeler, F. Schreiber, A. Bateman and S.R. Eddy 
**HMMER web server: 2015 update** ** **  
Nucleic Acids Research (2015) Web Server Issue 43:W30-W38. 
R.D. Finn, J. Clements and S.R. Eddy 
**HMMER web server: interactive sequence similarity searching 
**Nucleic Acids Research (2011) Web Server Issue 39:W29-W37. 
Eddy S.R. (2011) 
DOI: https://doi.org/10.1093/nar/gkr367  
**Accelerated Profile HMM Searches.**  
PLoS Computational Biology 7(10):e1002195 
Finn R.D., Attwood T.K., Babbitt P.C., Bateman A., Bork P., Bridge A.J., Chang H.Y., Dosztányi Z., El-Gebali S., Fraser M., Gough J., Haft D., Holliday G.L., Huang H., Huang X., Letunic I., Lopez R., Lu S., Marchler-Bauer A., Mi H., Mistry J., Natale D.A., Necci M., Nuka G., Orengo C.A., Park Y., Pesseat S., Piovesan D., Potter S.C., Rawlings N.D., Redaschi N., Richardson L., Rivoire C., Sangrador-Vegas A., Sigrist C., Sillitoe I., Smithers B., Squizzato S., Sutton G., Thanki N., Thomas P.D., Tosatto S.C., Wu C.H., Xenarios I., Yeh L.S., Young S.Y., Mitchell A.L. 
**InterPro in 2017-beyond protein family and domain annotations**  
Nucleic Acids Res Volume 45 (2017)D190-D199 
DOI: 10.1093/nar/gkw1107  
Jones P, Binns D, Chang HY, Fraser M, Li W, McAnulla C, McWilliam H, Maslen J, Mitchell A, Nuka G, Pesseat S, Quinn AF, Sangrador-Vegas A, Scheremetjew M, Yong SY, Lopez R, Hunter S. (2014) 
**InterProScan 5: genome-scale protein function classification.**  
Bioinformatics. 30(9) 1236-1240. 
DOI: 10.1093/bioinformatics/btu031.  
Quevillon E., Silventoinen V., Pillai S., Harte N., Mulder N., Apweiler R. and Lopez R. (2005) 
**InterProScan: protein domains identifier.**  
Nucleic Acids Res. 33:W116-20 
DOI: 10.1093/nar/gki442  
Zdobnov E.M. and Apweiler R. (2001) 
**InterProScan–an integration platform for the signature-recognition methods in InterPro.**  
Mistry J. (2007) 
**Predicting active site residue annotations in the Pfam database.**  
BMC Bioinformatics 8:298 
Käll L., Krogh A. and Sonnhammer E.L. (2004) 
**A combined transmembrane topology and signal peptide prediction method. 
**Journal of Molecular Biology 338(5):1027-36 
**R** ** ADAR **
Heger A. and Holm L. (2000) 
**Rapid automatic detection and alignment of repeats in protein sequences.**  
Proteins 41(2): 224-237 
Jonassen I, Collins JF, Higgins DG. (1995) 
**Finding flexible patterns in unaligned protein sequences.**  
Protein Sci. 4(8):1587-95. 
DOI: 10.1002/pro.5560040817  
Jonassen I. (1997) 
**Efficient discovery of conserved patterns using a pattern graph.**  
Comput Appl Biosci. 13(5):509-22. 
Burge SW, Daub J, Eberhardt R, Tate J, Barquist L, Nawrocki EP, Eddy SR, Gardner PP, Bateman A (2013) 
**Rfam 11.0: 10 years of RNA families.**  
Nucleic Acids Research 41:D226-32 
DOI: http://dx.doi.org/10.1093/nar/gks1005  
Nawrocki EP, Eddy SR (2013) 
**Infernal 1.1: 100-fold faster RNA homology searches. **  
Guerra-Assunção JA, Enright AJ (2010) 
**MapMi: automated mapping of microRNA loci**  
BMC bioinformatics 11 :133 
Li W., McWilliam H., Richart de la Torre A., Grodowski A., Benediktovich I., Goujon M., Nauche, S. and Lopez, R. (2010) 
** Non-redundant patent sequence databases with value-added annotations at two levels. **  
Nucleic Acids Research. 38 (Database issue): D52-D56 
PubMed:  19884134 
Pearson W.R, (1991) 
**Searching protein sequence libraries: comparison of the sensitivity and selectivity of the Smith-Waterman and FASTA algorithms.**  
Genomics 11 (3) :635-650 
Pearson W.R, (1990) 
**Rapid and sensitive sequence comparison with FASTP and FASTA**  
Methods in Enzymology 183 :63-98 
Pearson W.R. ,  Lipman D.J. (1988) 
** Comparative Study, research-article, Journal Article  
** Proceedings of the National Academy of Sciences of the United States of America 85 (8) :2444-2448 
Mackey A.J., Haystead T.A. and Pearson W.R. (2002) 
**Getting more from less: algorithms for rapid protein identification with multiple short peptide sequences.**  
Mol. Cell Proteomics 1(2):139-47. 
#### NCBI BLAST+  
Camacho, C., Coulouris, G., Avagyan, V., Ma, N., Papadopoulos, J., Bealer, K. and Madden T.L. (2009) 
**BLAST+: architecture and applications.**  
BMC Bioinformatics 10:421 
DOI:  10.1186/1471-2105-10-421  
Altschul, S.F., Gish, W., Miller, W., Myers, E.W. and Lipman, D.J. (1990) 
**Basic local alignment search tool.**  
Journal of Molecular Biology 215(3):403-10 
#### PSI BLAST  
Altschul S.F., Madden T.L., Schaffer A.A., Zhang J., Zhang Z., Miller W. and Lipman D.J. (1997) 
**Gapped BLAST and PSI-BLAST: a new generation of protein database search programs.**  
Nucleic Acids Res. 25(17):3389-402. 
Li W., McWilliam H., Goujon M., Cowley A., Lopez R. and Pearson W.R. (2012) 
**PSI-Search: iterative HOE-reduced profile SSEARCH searching.**  
### Help & Support
* No labels 
* Powered by Atlassian Confluence 6.13.3
* Printed by Atlassian Confluence 6.13.3
* Report a bug
{"serverDuration": 380, "requestCorrelationId": "30fdbdf833102462"}